test/E2E/Helpers.purs

```diff
@@ -0,0 +1,365 @@
+-}
+
+geroSign :: RunningExample -> Aff Unit
+geroSign (RunningExample { browser }) = do
+  pure unit
+
+-- | Wrapper for Page so it can be used in `shouldSatisfy`, which needs 'Show'
+-- | Doesn't show anything, thus 'NoShow'
+newtype NoShowPage = NoShowPage Toki.Page
+
+derive instance Newtype NoShowPage _
+
+instance Show NoShowPage where
+  show _ = "Toppoki.Page"
+
+newtype Selector = Selector String
+
+derive instance Newtype Selector _
+
+newtype Action = Action String
+
+derive instance Newtype Action _
+
+-- | Build a primitive jQuery expression like '$("button").click()' and evaluate it in Toki
+doJQ :: Selector -> Action -> Toki.Page -> Aff Foreign
+doJQ selector = doJQInd selector Nothing
+
+doJQ' :: Selector -> Action -> Toki.Page -> Aff Foreign
+doJQ' selector = doJQInd selector (Just 0)
+
+doJQInd :: Selector -> Maybe Int -> Action -> Toki.Page -> Aff Foreign
+doJQInd selector index action page = do
+  Toki.unsafeEvaluateStringFunction jq page
+  where
+  jq :: String
+  jq = "$('" <> unwrap selector <> "')" <> indexStr <> "." <> unwrap action
+
+  indexStr :: String
+  indexStr = maybe "" (\i -> "[" <> show i <> "]") index
+
+click :: Action
+click = wrap "click()"
+
+enable :: Action
+enable = wrap "prop(\"disabled\", false)"
+
+setAttr :: String -> String -> Action
+setAttr attr value = wrap $ "attr(" <> jqStr attr <> ", " <> value <> ")"
+
+button :: Selector
+button = wrap "button"
+
+buttonWithText :: String -> Selector
+buttonWithText text = wrap $ "button:contains(" <> text <> ")"
+
+inputType :: String -> Selector
+inputType typ = wrap $ "input[type=\"" <> typ <> "\"]"
+
+textarea :: Selector
+textarea = wrap "textarea"
+
+byId :: String -> Selector
+byId = wrap <<< ("#" <> _)
+
+password :: Selector
+password = wrap ":password"
+
+jqStr :: String -> String
+jqStr str = "\"" <> str <> "\""
+
+jqSet :: String -> String -> Action
+jqSet what value = wrap $ what <> "(" <> value <> ")"
+
+setText :: String -> Action
+setText = jqSet "text" <<< jqStr
+
+setValue :: String -> Action
+setValue = jqSet "val" <<< jqStr
+
+trigger :: String -> Action
+trigger event = wrap $ "trigger(" <> jqStr event <> ")"
+
+clickButton :: String -> Toki.Page -> Aff Unit
+clickButton buttonText = void <$> doJQ (buttonWithText buttonText) click
+
+injectJQueryAll :: String -> Toki.Browser -> Aff (Array Toki.Page)
+injectJQueryAll jQuery browser = do
+  pages <- Toki.pages browser
+  void $ for pages $ \page -> do
+    (alreadyInjected :: Boolean) <-
+      unsafeFromForeign <$>
+        Toki.unsafeEvaluateStringFunction "typeof(jQuery) !== 'undefined'" page
+    unless alreadyInjected $ void $ Toki.unsafeEvaluateStringFunction jQuery
+      page
+  pure pages
+
+reactSetValue :: Selector -> String -> Toki.Page -> Aff Unit
+reactSetValue selector value page = void
+  $ flip Toki.unsafeEvaluateStringFunction page
+  $ fold
+      [ -- https://stackoverflow.com/questions/23892547/what-is-the-best-way-to-trigger-onchange-event-in-react-js
```

Comment:
This should be moved to a top-level doc comment I think, it's a pretty important explanation